src/main/java/dev/andrylat/vomelianchuk/card_number_operations/CardNumberValidator.java

```diff
@@ -0,0 +1,51 @@
+package dev.andrylat.vomelianchuk.card_number_operations;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class CardNumberValidator {
+    private static final String HIATUS_SEPARATOR = "\\s";
+    private static final int LENGTH_CARD_NUMBER = 16;
+    private static final String ERROR_NULL_CARD_NUMBER = "Error receiving null object instead of card number";
+    private static final String ERROR_LENGTH_CARD = "Length should be 16 symbols";
+    private static final String ERROR_ONLY_DIGITS = "Number should contain only digits";
+
+    private final List<String> errors = new ArrayList<>();
+
+    public void validate(String cardNumber) throws CardValidationException {
```

Comment:
Nope, nope, nope! Very big mistake.

Or you use state in your class and you inject all necessary data to the constructor (as it was as I remember), and after it you call method validate(), getErrors(), etc.

OR you avoid using state and it means that you method receive all necessary information and return result (or return results, or throw exception if needed). BUT in this case YOU SHOULD NOT USE STATE in the class (line 13), because your code is not thread safe and you always must remember about cleanup. Our classes must be as simple as possible. I don't need to think more that it is needed during usage of class.

Reply:
Fixed here: 7a87509